ui/src/app/views/confirm-delete-modal/component.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  Component, Input,
+} from '@angular/core';
+import * as React from 'react';
+
+import {
+  capitalize,
+  decamelize,
+  ReactWrapperBase
+} from 'app/utils';
+
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+
+import {
+  Button
+} from 'app/components/buttons';
+
+export interface ConfirmDeleteModalProps {
+  deleting: boolean;
+  closeFunction: Function;
+  resourceType: string;
+  receiveDelete: Function;
+  resource: {name: string};
+}
+
+export interface ConfirmDeleteModalState {
+  loading: boolean;
+}
+
+export class ConfirmDeleteModal
+    extends React.Component<ConfirmDeleteModalProps, ConfirmDeleteModalState> {
+  state: ConfirmDeleteModalState;
+  props: ConfirmDeleteModalProps;
```

Comment:
You don't need to declare `state` and `props` types separately here. They're automatically inferred from the declarations in the angle brackets above.